llvm/test/compiler-course/grudzin_k_pureF_test/grudzin_k_pureF.ll

```diff
@@ -0,0 +1,47 @@
+; RUN: opt -load-pass-plugin %llvmshlibdir/MarkPurePass_Grudzin_Konstantin_FIIT1_LLVM_IR%pluginext -passes="mark-pure-pass" -S %s | FileCheck %s
+
+; Данный комментарий появился здесь для того, чтобы не забыть это на будующей защите:
+; Почему я использую #0 (т.к так помечает эти функции IR, в строке компиляции видно, что он ставит после объявления #0 и добавляет в конце attributes #0 = {"pure"})
```

Comment:
This is a good way to check, but you need to check the function name to which this attribute was attached, and also you need to add a check for what attribute `#0` actually is (basically, add the check for `attributes #0 = {"pure"} `), because it may change after pass application